### PR TITLE
custom objects implemented -target base changed

### PR DESCRIPTION
**Pull request by santiago8781**

### CODE REVIEW

File: freshservice/custom_objects.go [hunk 1]
```diff
@@ -18,34 +16,64 @@ type CustomObjectService struct {
...
-	//Meta
+// CustomObjectRecord contains a single custom object record
+type CustomObjectRecord struct {
+	CreatedAt      time.Time `json:"bo_created_at,omitempty"`
+	CreatedBy      CreatedBy `json:"bo_created_by,omitempty"`
+	DisplayId      int       `json:"bo_display_id,omitempty"`
+	UpdatedAt      time.Time `json:"bo_updated_at,omitempty"`
+	UpdatedBy      UpdatedBy `json:"bo_updated_by,omitempty"`
+	DataSource     string    `json:"data_source_name,omitempty"`
+	DataSourceType string    `json:"data_source_id,omitempty"`
```
Comment (marcoshuck) [inline freshservice/custom_objects.go]:
```suggestion
```

File: freshservice/custom_objects.go [hunk 2]
```diff
@@ -18,34 +16,64 @@ type CustomObjectService struct {
...
+}
+
+type CreatedBy struct {
```
Comment (marcoshuck) [inline freshservice/custom_objects.go:46]: Missing documentation

File: freshservice/custom_objects.go [hunk 3]
```diff
@@ -18,34 +16,64 @@ type CustomObjectService struct {
...
+	Title       string   `json:"title,omitempty"`
+	Description string   `json:"description,omitempty"`
+	Fields      []Fields `json:"fields,omitempty"`
+	Meta        int      `json:"total_records_count,omitempty"`
 }
 
-// CustomObjectWrapper contains Details of a CustomObject
-type CustomObjectWrapper struct {
-	Details CustomObject `json:"custom_object"`
+type Fields struct {
```
Comment (marcoshuck) [inline freshservice/custom_objects.go:28]: Missing documentation

File: freshservice/custom_objects.go [hunk 4]
```diff
@@ -18,34 +16,64 @@ type CustomObjectService struct {
 	client *Client
 }
 
-// CustomObjects contains Collection an array of CustomObject
-type CustomObjects struct {
-	Collection []CustomObject `json:"custom_objects"`
+type CustomObject struct {
```
Comment (marcoshuck) [inline freshservice/custom_objects.go:19]: Missing documentation

File: freshservice/custom_objects.go [hunk 5]
```diff
@@ -18,34 +16,64 @@ type CustomObjectService struct {
...
+type CreatedBy struct {
+	Id    int    `json:"id,omitempty"`
+	Value string `json:"value,omitempty"`
 }
 
-type LastUpdatedBy struct {
-	Email  string `json:"email,omitempty"`
-	ID     int    `json:"id,omitempty"`
-	System string `json:"name,omitempty"`
+type UpdatedBy struct {
```
Comment (marcoshuck) [inline freshservice/custom_objects.go:51]: Missing documentation

File: freshservice/custom_objects.go [hunk 6]
```diff
@@ -18,34 +16,64 @@ type CustomObjectService struct {
...
+	Collection []CustomObject `json:"custom_objects"`
+}
+
+//CustomObjectRecordWrapper contains Details of a CustomObjectRecords
+type CustomObjectRecordWrapper struct {
+	Data CustomObjectRecord `json:"data,omitempty"`
+}
+
+type CustomObjectRecordUpdate struct {
+	CustomObjectRecord CustomObjectRecord `json:"custom_object,omitempty"`
```
Comment (marcoshuck) [inline freshservice/custom_objects.go:74]: I'm surprised we don't need to wrap this field, does he inner object have a `data` field?

File: freshservice/custom_objects.go (hunk 5 above)
Comment (santiago8781) [reply freshservice/custom_objects.go:51]: will do

File: freshservice/custom_objects.go (hunk 4 above)
Comment (santiago8781) [reply freshservice/custom_objects.go:19]: will do

File: freshservice/custom_objects.go (hunk 3 above)
Comment (santiago8781) [reply freshservice/custom_objects.go:28]: will do

File: freshservice/custom_objects.go (hunk 2 above)
Comment (santiago8781) [reply freshservice/custom_objects.go:46]: will do

File: freshservice/custom_objects.go (hunk 6 above)
Comment (santiago8781) [reply freshservice/custom_objects.go:74]: we have a CustomObjectRecordWrapper

**Pull request merged**